active/0000-removeAddonFromProject.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 2015-05-18
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+Allow addons to clean up after their removal and to remove up other conflicting packages on install.
+
+# Motivation
+
+Managing addons becomes much simpler if the author doesn't need to understand dependencies that the author has had to account for.
+This has been an issue with [ember-cli-mocha](https://github.com/switchfly/ember-cli-mocha) where qunit has conflicted.
+
+# Detailed design
+
+In an addons blueprint the author could write a `afterUninstall` hook to remove all it's dependencies:
+```
+  afterInstall: function() {
+    var addonContext = this;
+
+    return this.addBowerPackagesToProject([
+      {
+        name: 'ember-mocha',
+        target: '~0.6.2'
+      },
+      {
+        name: 'ember-cli/ember-cli-test-loader',
+        target: '0.1.3'
+      },
+      {
+        name: 'ember-cli/ember-cli-shims',
+        target: '0.0.3'
+      }
+    ]).then(function() {
+      if ('removeAddonFromProject' in addonContext) {
+        return addonContext.removeAddonFromProject('ember-cli-qunit');
+      }
+    });
+  },
+  afterUninstall: function () {
+     this.removeBowerPackagesFromProject([
```

Comment:
Well for example if I can have some way to express what my dependencies are as a list I think this might be much easier to manage for installing, warning about conflicts and potentially allowing later to prune the dependencies if the user wants.

Behind the scenes this could just register diagnostic checks without having to express all the syntax etc.

So by adding the following on install it could add the user added packages and then warn when using ember doctor if the user has qunit installed, it could hook into the same API behind the scenes just making it simpler to implement.
```
  dependencies: function () {
    return {
      required: {
        bower: [
          {
            name: 'ember-mocha',
            target: '~0.6.2'
          },
          {
            name: 'ember-cli/ember-cli-test-loader',
            target: '0.1.3'
          },
          {
            name: 'ember-cli/ember-cli-shims',
            target: '0.0.3'
          }
        ]
      },
      conflicts: {
        addons: [
          'ember-cli-qunit'
        ]
      }
    }
  }
```

I think this might also solve the issue of addon dependencies a little neater than peerDependencies (Which is going in 3.0.0 if I remember correctly) by potentially allowing addons not to have to install into the package.json.